clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2261,6 +2262,12 @@ class UnsafeBufferUsageReporter : public UnsafeBufferUsageHandler {
         // note_unsafe_buffer_operation doesn't have this mode yet.
         assert(!IsRelatedToDecl && "Not implemented yet!");
         MsgParam = 3;
+      } else if (isa<MemberExpr>(Operation)) {
+        // note_unsafe_buffer_operation doesn't have this mode yet.
+        assert(!IsRelatedToDecl && "Not implemented yet!");
+        auto ME = dyn_cast<MemberExpr>(Operation);
+        name = ME->getMemberDecl()->getName();
```

Comment:
I suspect that this one will crash in case of anonymous fields. That said, I also don't think it's even possible to put an attribute on an anonymous field, so we're probably fine.

However, with diagnostics specifically, the tradition isn't to put `name` into the `Diag()` stream; the tradition is to put the decl itself, directly, by pointer into the Diag() stream!

I.e.
```c++
const Decl *D = ME->getMemberDecl();
...
S.Diag(Loc, diag::warn_unsafe_buffer_operation) << MsgParam << D << Range;
```
This way not only it reacts to anonymous objects correctly (I think), but also it automatically puts quotes around the name, which is something we currently forget to do :)

Reply:
Let's add a tiny test that we are not introducing the crash.

Reply:
Sounds good, I will use the Decl and also add a test. I wasn't aware that it is unusual to put a field name in there. 

Side note: we should probably also provide the Decl, when the attribute is added to method overloads. Perhaps that needs to be a separate PR?

Reply:
> Perhaps that needs to be a separate PR?

It shouldn't block this PR but it'd indeed be very nice to have either way.